code/modules/integrated_electronics/subtypes/manipulation.dm

```diff
@@ -79,7 +78,7 @@
 		to_chat(user, "<span class='notice'>There's no weapon to remove from the mechanism.</span>")
 
 /obj/item/integrated_circuit/manipulation/weapon_firing/do_work()
-	if(!installed_gun)
+	if(!installed_gun || !installed_gun.handle_pins())
```

Comment:
Have you tested it?Are you sure , it won't glitch without user?

Reply:
yes i even made a comment and added istypes for if it's null.